.github/workflows/script.py

```diff
@@ -0,0 +1,143 @@
+import os
+import yaml
+import sys
+
+os.makedirs('plugins/OpenShift', exist_ok=True)
+
+if os.path.exists("index.yaml"):
+    file = open("index.yaml","r")
+    not_present = 1
+    index = yaml.safe_load(file)
+    for plugin in index['plugins']:
+        if plugin['name'] == 'OpenShiftPlugin':
+            not_present = 0
+            break
+    if not_present:
+        index['plugins'].append({"name": "OpenShiftPlugin", "path": "https://github.com/%s/crane-plugins/raw/main/plugins/OpenShift/index.yaml"%sys.argv[2]})
+    file = open("index.yaml","w")
+    yaml.dump(index, file)
+    file.close()
+
+else:
```

Comment:
Personally, I think we should fail hard if there is no PluginIndex at the root of the project.